File: android/app/src/main/kotlin/pl/kwasow/sunshine/data/Memory.kt
package pl.kwasow.sunshine.data

import kotlinx.serialization.Serializable
import java.time.LocalDate
import java.time.format.DateTimeFormatter
import java.time.format.FormatStyle

@Serializable
data class Memory(
    val id: Int,
    val startDate: String,
    val endDate: String?,
    val title: String,
    val description: String,
    val photo: String?,
) {
    companion object {
        val dateFormatter = DateTimeFormatter.ofLocalizedDate(FormatStyle.LONG)
    }

    // ====== Public methods
    fun getLocalStartDate(): LocalDate? {
        return try {
            LocalDate.parse(startDate)
        } catch (e: Exception) {
            null
        }
    }

    fun getStringDate(): String {
        val localDate = getLocalStartDate() ?: return startDate

        return dateFormatter.format(localDate)
    }
}

File: backend/src/entities/memory.php
<?php

class Memory implements JsonSerializable
{
    private $_id;
    private $_startDate;
    private $_endDate;
    private $_title;
    private $_description;
    private $_photo;

    public function __construct(
        $id,
        $startDate,
        $endDate,
        $title,
        $description,
        $photo
    ) {
        $this->_id = $id;
        $this->_startDate = $startDate;
        $this->_endDate = $endDate;
        $this->_title = $title;
        $this->_description = $description;
        $this->_photo = $photo;
    }

    public function jsonSerialize()
    {
        return [
        'id' => $this->_id,
        'startDate' => $this->_startDate,
        'endDate' => $this->_endDate,
        'title' => $this->_title,
        'description' => $this->_description,
        'photo' => $this->_photo
        ];
    }

}

?>
